[RHDEVDOCS-6657] Release notes for Pipelines 1.20

ochromy opened the pull request:
Version(s):
- pipelines-docs-1.20

Issue:
- https://issues.redhat.com/browse/RHDEVDOCS-6657

Link to docs preview:
- [Pipelines 1.20 release notes](https://98437--ocpdocs-pr.netlify.app/openshift-pipelines/latest/release_notes/op-release-notes-1-20.html)

QE review:
- [x] QE has approved this change.

Additional info:

khrm commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,195 @@
+#...
+----
++
+If set to `true`, runs are not stored until they are complete. If `false`, runs are stored upon creation and continuously upserted until deletion (current default). Using `true` helps improve performance.
```
continuously upserted whenever there's a change.

Probably, we can mention that this will become the default behaviour in coming releases. So you might need to explicitly set it to false.

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,195 @@
+.{pac}
+* {pac} synchronizes the status of running `PipelineRun` on git providers when managed by `Kueue`, a Kubernetes job-queuing system. This enhancement helps developers receive feedback on their pull requests faster through comments and status checks.
+
+* {pac} supports passing webhook parameters in the JSON POST body, while maintaining compatibility with URL query parameters. Using the request body for sensitive information reduces exposure in logs and improves security.
```
@ochromy I think it should contain example. I don't know how to format it you can a look at that.

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,195 @@
+
+* {pac} supports detailed logging for GitHub API calls, providing insights into API interactions, durations, and rate-limiting. By setting the controller's log level to 'debug', you can troubleshoot complex issues more efficiently. This enhancement refactors code to instrument GitHub calls and affects all types of GitHub API calls within the provider.
+
+* {pac} adds new `auto-configure-new-github-repo` setting in to consolidate repository creation within a single namespace, streamlining management for those with multiple repositories. This helps in creating more organized and manageable environment by autoconfiguring repository names, eliminating the potential chaos caused by multiple namespaces for each repository. When using this new setting,  ...
```
```suggestion
* {pac} adds new option in `auto-configure-new-github-repo` feature setting to consolidate repository creation within a single namespace, streamlining management for those with multiple repositories. This helps in creating more organized and manageable environment by autoconfiguring repository names, eliminating the potential chaos caused by multiple namespaces for each repository. Now, you can configure your custom repository name by using the `auto-configure-repo-repository-template` setting.
 
* see example:
 
* auto-configure-new-github-repo: "true"
* auto-configure-repo-namespace-template: "all-repositories-ns"
* auto-configure-repo-repository-template: "{{repo_owner}}-{{repo_name}}-cr"

when you configure these settings, {pac} will create Repository CR for all the newly created GitHub repositories in `all-repositories-ns` namespace. for exmaple if you create a repository on GitHub with name `org/test-repo` a Repository CR will be created in `all-repositories-ns` namespace like this `org-test-repo-cr`.
```

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,195 @@
+[id="pipelines-breaking-changes-1-20-0_{context}"]
+=== Breaking changes
+
+* {pac} automatically moves from using the deprecated {tekton-hub} to {artifact-hub} for task resolution. After the migration some task references using short version pins, such as `0.2`, will not resolve correctly. You must update these pins to the full semantic version format, such as `0.2.0`, to ensure correct task resolution.
```
```suggestion
* {pac} automatically moves from using the deprecated {tekton-hub} to {artifact-hub} for pipeline, task resolution. After the migration some task references using short version pins, such as `0.2`, will not resolve correctly. You must update these pins to the full semantic version format, such as `0.2.0`, to ensure correct task resolution.
```

divyansh42 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,195 @@
+
+* Before this update, the `git-clone` task failed with a `No such remote 'origin'` error messgae if the `origin` remote was missing from the repository. With this update, the task automatically adds the `origin` remote to the repository configuration, ensuring correct setup and successful cloning.
+
+* Before this update, streaming live logs for a PipelineRun or TaskRun with the {tekton-results} CLI required the `-f` flag. Attempts to query logs without the flag resulted in errors, and status messages for running resources were unclear. With this update, the CLI streams logs without requiring the `-f` flag, and provides clearer status messages when querying logs for running `PipelineRuns` and ...
```
For some reason, this is coming inside the "pipelines" section. Which is not correct.
Also, this point is conveying the wrong information. Correct point should be:
```
Before this update, users querying logs for a running PipelineRun or TaskRun in OPC Results received an unclear error message. With this update, the message clearly indicates that logs cannot be retrieved while the run is still in progress.
```

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,195 @@
+.{pac}
+* Before this update, structured logs for {pac} lacked detailed source repository information from the initiating webhook request. With this update, logs for {pac} include complete source repository details, making it easier for operators to identify user-reported issues.
+
+* Before this update, {pac} failed immediately when resource quotas were exceeded, canceling the run and interrupting user workflows. With this update, {pac} controllers retry and automatically rerun if resources become available, reducing unnecessary cancellations and improving pipeline reliability.
```
@ochromy thanks! actually it is not PaC bug it was `tektoncd/pipeline` component bug so you should move it to there

anithapriyanatarajan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,225 @@
+.Pruner
+* Before this update, the event-based pruner failed to process `PipelineRun` objects when the `historyLimit` setting was set to `0`, leaving them in a running state. With this update, the pruner handles `PipelineRuns` with `historyLimit` set to `0`, improving resource management.  
+
+* Before this update, the pruner could panic during upgrades if a setting was `nil`. With this update, the pruner handles `nil` values correctly and avoids crashes.
```
I think we could avoid this since this faced by non openshift users

anithapriyanatarajan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,225 @@
+* Before this update, {pac} failed immediately when resource quotas were exceeded, canceling the run and interrupting user workflows. With this update, {pac} controllers retry and automatically rerun if resources become available, reducing unnecessary cancellations and improving pipeline reliability.
+
+.Pruner
+* Before this update, the event-based pruner failed to process `PipelineRun` objects when the `historyLimit` setting was set to `0`, leaving them in a running state. With this update, the pruner handles `PipelineRuns` with `historyLimit` set to `0`, improving resource management.  
```
I am sorry, is this a release bug? I dont remember where this is coming from?

anithapriyanatarajan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,225 @@
+#...
+----
+
+* Before this update, the controller used a fixed thread count, limiting concurrency. With this update, you can override `threads-per-controller` by setting the `THREADS_PER_CONTROLLER` environment variable, allowing finer control over controller concurrency.
```
Please include the correct component controller to which this fix was done? Also is there any max limit?

anithapriyanatarajan reviewed (changes requested):
updated a few comments

mbpavan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,225 @@
+* You can choose whether to install {tekton-triggers} when deploying {pipelines-shortname} through the Operator. This enhancement provides finer control over component installation, allowing flexibility for environments where triggers are managed separately.
+
+.{tekton-results}
+* {tekton-results} supports PostgreSQL version 17.
```
please remove {tekton-results} supports PostgreSQL version 17. - its not needed

mbpavan commented on modules/op-tkn-pipelines-compatibility-support-matrix.adoc:
```diff
@@ -23,13 +23,14 @@ GA:: General Availability
-|1.15 | 0.59.x | 0.27.x | 0.37.x | 0.20.x (GA) | 1.17.x (TP) | 0.27.x (GA) | 0.10.x (TP) | 0.2.x (TP) | 4.14, 4.15, 4.16 | GA
+|1.18 | 0.68.x | 0.31.x | 0.40.x | 0.24.x (GA) | 1.20.x (TP) | 0.33.x (GA) | 0.14.x (GA) | 0.5.x (TP) | 4.15, 4.16, 4.17, 4.18 | GA
 
-|1.14 | 0.56.x | 0.26.x | 0.35.x | 0.20.x (GA) | 1.16.x (TP) | 0.24.x (GA) | 0.9.x (TP) | NA | 4.12, 4.13, 4.14, 4.15, 4.16 | GA
```
I think we should confirm this with @savitaashture (she’ll be back on Monday), or check with Misha in the meantime. For 1.19, I see 1.16, 1.17, 1.18, and 1.19 included.

mbpavan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,225 @@
+
+* * With this update, you can independently control whether the {pipelines-shortname} Operator creates Role-Based Access Control (RBAC) resources and Trusted Certificate Authority (CA) bundle ConfigMaps in your cluster. Two new options in the `TektonConfig` custom resource (CR) allow you to configure this behavior:  
++
+** `createRbacResource`: Enables or disables creation of RBAC resources.  
```
for new features, we need to specify under particular component similar to 1.19. For example - this should be under operator section

mbpavan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,225 @@
+
+* You can use relative paths to reference tasks from within a remote `Pipeline` definition. The {pac} resolver automatically builds the full URL for the task based on the location of the remote `PipelineSpec` definition. This improves pipeline organization and portability by keeping tasks and pipelines co-located, which is especially useful when working with tagged releases or branches.
+
+.Helm
```
we shoudnt have section for helm, it should be included in the component section, for this its on operator

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,220 @@
+* All containers in the `tekton-results` deployment have `readOnlyRootFilesystem` setting enabled, improving security and compliance.
+
+.{pac}
+* {pac} synchronizes the status of running `PipelineRun` on git providers when managed by `Kueue`, a Kubernetes job-queuing system. This enhancement helps developers receive feedback on their pull requests faster through comments and status checks.
```
@anithapriyanatarajan thank you for bringing this in attention, Yeah this is only for Konflux and I think should not be gone into release notes. @ochromy can you please remove this?

srivickynesh commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,236 @@
+
+* Before this update, git resolver deployments could leave zombie git processes running, consuming resources. With this update, resolvers  use a `tini`-based image and entrypoint to actively clean up subprocesses.
+
+* * Before this update, upgrading {pipelines-shortname} could cause errors when updating metadata, such as finalizers, on completed `PipelineRuns` or `TaskRuns` due to specification drift. With this update, the issue is fixed.
```
```suggestion
* Before this update, upgrading {pipelines-shortname} could cause errors when updating metadata, such as finalizers, on completed `PipelineRuns` or `TaskRuns` due to specification drift. With this update, the issue is fixed.
```

divyansh42 reviewed (changes requested):
Just a couple of minor suggestions, Just wanted to make sure that we don't generalise and be specific about the CLI which we are talking about. 
As we are adding OPC and tkn both in the same section

divyansh42 reviewed (commented):
LGTM

manthinasai commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,238 @@
+* Before this update, auto-merge was blocked if an unauthorized user opened a pull request (PR) and the target branch did not contain a `.tekton` directory. {pac} created a pending check that remained indefinitely, even after a repository admin approved the CI run with the `/ok-to-test` GitOps comment. With this update, the check is updated correctly after approval, allowing auto-merge to proceed ...
+
+.CLI
+* Before this update, `tkn` CLI reading pod logs from already deleted pods could cause a panic, leading to application crashes. With this update, deleted pod scenarios are handled properly, preventing crashes when reading logs.
```
@divyansh42 Shouldn't the CLI reference pointed to opc ?

mbpavan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,234 @@
+* Before this update, auto-merge was blocked if an unauthorized user opened a pull request (PR) and the target branch did not contain a `.tekton` directory. {pac} created a pending check that remained indefinitely, even after a repository admin approved the CI run with the `/ok-to-test` GitOps comment. With this update, the check is updated correctly after approval, allowing auto-merge to proceed ...
+
+.CLI
+* Before this update, CLI reading pod logs from already deleted pods could cause a panic, leading to application crashes. With this update, deleted pod scenarios are handled properly, preventing crashes when reading logs.
```
please change tkn to opc

mbpavan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,234 @@
+.CLI
+* Before this update, CLI reading pod logs from already deleted pods could cause a panic, leading to application crashes. With this update, deleted pod scenarios are handled properly, preventing crashes when reading logs.
+
+* Before this update, the CLI `PipelineRunPending` status was displayed with incorrect coloring, making it harder to distinguish from other statuses. With this update, the Pending status color is correct for improved visual clarity.
```
please change tkn to opc

mbpavan commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,234 @@
+
+* Before this update, the CLI `PipelineRunPending` status was displayed with incorrect coloring, making it harder to distinguish from other statuses. With this update, the Pending status color is correct for improved visual clarity.
+
+* Before this update, CLI following logs could trigger a deadlock, causing the application to hang. With this update, the deadlock scenario in the log following functionality is fixed, ensuring reliable log streaming.
```
please change tkn to opc

srivickynesh commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,278 @@
+[id="pipelines-kn-issues-1-20-0_{context}"]
+=== Known issues
+
+* Using `tkn pr logs` in the OpenShift namespace may fail with repeated `Failed to list objects from openshift namespace` errors for both admin and non-admin users. 
```
```suggestion
* Using `opc pr logs` in the OpenShift namespace may fail with repeated `Failed to list objects from openshift namespace` errors for both admin and non-admin users. 
```

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,281 @@
+Using the request body for sensitive information reduces exposure in logs and improves security. For more details, see link:https://pipelinesascode.com/docs/guide/incoming_webhook/#recommended-post-json-body-method[Recommended (POST JSON body) method].
+
+Detailed logging for GitHub API calls::
+* {pac} supports detailed logging for GitHub API calls, providing insights into API interactions, durations, and rate-limiting. By setting the controller log level to 'debug', you can troubleshoot complex issues more efficiently. This enhancement refactors code to instrument GitHub calls and affects all types of GitHub API calls within the provider. For more details, see link:https://pipelinesasc ...
```
@ochromy can you refer upstream docs in downstream docs? e.g. https://pipelinesascode.com

ochromy commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,281 @@
+
+* Before this update, the `git-clone` task failed with a `No such remote 'origin'` error messgae if the `origin` remote was missing from the repository. With this update, the task automatically adds the `origin` remote to the repository configuration, ensuring correct setup and successful cloning.
+
+* Before this update, {pac} failed immediately when resource quotas were exceeded, canceling the run and interrupting user workflows. With this update, {pac} controllers retry and automatically rerun if resources become available, reducing unnecessary cancellations and improving pipeline reliability.
```
https://issues.redhat.com/browse/SRVKP-4165 - "PaC failing" in the description

ochromy replied:
But it is not in the `.{pac}` section og the fixed issues. Should I remove the mention of PAC specifically?

zakisk replied:
You should move it to pipeline

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,281 @@
+* The `git-clone` task downloaded from {tekton-hub} displays version `0.9` instead of `0.10`. This occurs because the system sorts version strings lexicographically rather than using semantic versioning.
+
+.{pac}
+* When upgrading to 1.20, the `hub_catalog_name` variable value is set to `tekton` by default and remains set to `tekton` after the upgrade. As a result, the system does not fetch the `git-clone` task from the correct catalog on {artifact-hub}.
```
but I still think that this shouldn't be included

zakisk commented:
@ochromy let me know you've done

srivickynesh commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,286 @@
++
+[source, terminal]
+----
+ `cmd - oc patch configmap pipelines-as-code -n openshift-pipelines --type=json -p='[{"op": "remove", "path": "/data/hub-catalog-name"}]'`
```
```suggestion
 oc patch configmap pipelines-as-code -n openshift-pipelines --type=json -p='[{"op": "remove", "path": "/data/hub-catalog-name"}]'
```

gabriel-rh reviewed (approved):
LGTM - some minor comments

srivickynesh commented on release_notes/op-release-notes-1-20.adoc:
```diff
@@ -0,0 +1,41 @@
+
+//Additional resources 1.20
+.Additional resources
+* For more information on the new `buildah-ns` taks, see xref:../secure/using-buildah-ns-tekton-task.adoc#op-differences-between-buildah-buildah-ns-tasks_using-buildah-ns-tekton-task[Differences between `buildah` and `buildah-ns` tasks].
```
Differences between `buildah` and `buildah-ns` tasks
hyperlink is broken

srivickynesh reviewed (commented):
lgtm

ochromy commented:
First three additional resources are not working in the preview, because the files are not present on this PRs branch, but after merging they should work fine. 
- In case the merge is impossible with prow failing, I'll either comment them out now and readd afterwards, or rebase the PR (this should be the easier of the two, but I'll have to take a look at the process after posting this).
OR 
- If we can merge even with the preview build failing, then the files should be on our main (or at least on pipelines-docs-1.20) and thus should work - we can test this afterwards in the Pantheon preview. 
@Dhruv-Soni11 WDYT? I'll still try the rebase, just want to know if it'd allow the merge without the portal validation.

ochromy committed:
[RHDEVDOCS-6657] Release notes for Pipelines 1.20

[RHDEVDOCS-6657] Release notes for Pipelines 1.20

fix rbac ref no1

merge review no1

merge review no2

merge review no3

merge review no4

merge review no5

merge review no6

zakisk commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+include::snippets/technology-preview.adoc[]
+
+New dynamic pull_request_number variable for push events::
+* Pull requests that trigger a push event automatically populate a new {pac} dynamic `pull_request_number` variable, allowing for a clear reference to the specific pull request associated with the push event and improving traceability. For more details, see "Additional resources".
```
```suggestion
* Pull requests that trigger a push event automatically populates a new {pac} dynamic `pull_request_number` variable, allowing for a clear reference to the specific pull request associated with the push event and improving traceability. For more details, see "Additional resources".
```

vdemeester commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+
+* Before this update, the pipeline builder fetched only a limited number of tasks from {artifact-hub}, resulting in some tasks not being available. This resulted in an incomplete pipeline creation. With this update, the interface fetches the complete list of tasks from {artifact-hub} on search, showing all available tasks in the pipeline builder.
+
+* Before this update, running `fbc-fips-check-oci-ta` tasks during git cloning caused temporary resource unavailability, stalling the git resolver and preventing the resolution of `ResolutionRequests`. This resulted in a build failure with the following error message: `cannot fork() for remote-https: Resource temporarily unavailable`. With this update, the issue is fixed.
```
This task is konflux specific isn't it ? For a customer that doesn't know anything about konflux, this entry is a bit confusing imo.

ochromy replied:
Should we rewrite it as:
```suggestion
* Before this update, running the Konflux-specific `fbc-fips-check-oci-ta` tasks during git cloning caused temporary resource unavailability, stalling the git resolver and preventing the resolution of `ResolutionRequests`. This resulted in a build failure with the following error message: `cannot fork() for remote-https: Resource temporarily unavailable`. With this update, the issue is fixed.
```

vdemeester commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+#...
+----
+
+* Before this update, the controllers used a fixed thread count, limiting concurrency. With this update, you can override `threads-per-controller` by setting the `THREADS_PER_CONTROLLER` environment variable, allowing finer control over controller concurrency.
```
Is it on a specific controller ? I thought tektoncd/pipeline supported this for a long time already.

vdemeester commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+
+* Before this update, git resolver shell-outs sometimes failed to inherit environment variables, breaking environment-driven authentication or configuration. With this update, the resolver passes the pod environment correctly to all git subprocesses.
+
+* Before this update, git resolver deployments could leave zombie git processes running, consuming resources. With this update, resolvers  use a `tini`-based image and entrypoint to actively clean up subprocesses.
```
@athorp96 isn't `tini` based only upstream ?

aThorp96 replied:
`tini` is [used downstream](https://github.com/openshift-pipelines/tektoncd-pipeline/blob/60a1d516229066f70233c56392a66cbc6a16e618/.konflux/dockerfiles/resolvers.Dockerfile#L54) as well, but it's not a `tini`-based image. It's not new to 1.20 though. Both of these changes were introduced in 1.19: [pod environment in git](https://github.com/openshift-pipelines/tektoncd-pipeline/blob/0a7000aba8ab139c014af62b191890818ca9ab82/upstream/pkg/resolution/resolver/git/repository.go#L122) and [tini entrypoint](https://github.com/openshift-pipelines/tektoncd-pipeline/blob/0a7000aba8ab139c014af62b191890818ca9ab82/.konflux/dockerfiles/resolvers.Dockerfile#L53)

vdemeester reviewed (commented):
Overall looks good, just few comments/questions

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+* {pipelines-shortname} adds and applies the `preferredDuringSchedulingIgnoredDuringExecution` pod anti-affinity rule to controller replicas, except for the {tekton-chains} controllers. In High Availability (HA) setups, this rule distributes replicas across different nodes instead of scheduling them on the same node, improving resiliency, load balancing, and availability. No additional configurat ...
+
+New buildah-ns task for improved container build security::
+* {pipelines-shortname} adds the `buildah-ns` task. It improves container build security with user namespace isolation and remains compatible with the existing `buildah` task. For more details, see "Additional resources".
```
For more details, see "Additional resources". -> For more information, see the <xyz> section in Additional Resources.

ochromy replied:
[lead-in sentences](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentences ) but I can change _details_ to _information_

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+* Containers in the {pipelines-shortname}, {tekton-results}, {tekton-chains}, and Manual Approval Gate deployments have `readOnlyRootFilesystem` setting enabled by default, improving security and compliance.
+
+Tasks display friendly names in the web console::
+* Tasks in the {OCP} web console now use the displayName property instead of technical task names, improving readability and making tasks easier to search within the {pipelines-title} interface.
```
displayName -> `displayName`

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+
+.Pruner
+
+Event-based pruner configurable in TektonConfig CR::
```
TektonConfig -> `TektonConfig`?

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+.Pruner
+
+Event-based pruner configurable in TektonConfig CR::
+* You can enable and configure the event-based `tektonpruner` directly in the `TektonConfig` custom resource (CR). Observability enhancement is also included with the new metrics specific to pruner. 
```
You can enable and configure the event-based tektonpruner in the TektonConfig custom resource (CR). This update also adds observability enhancements with new pruner-specific metrics.

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+.{tekton-triggers}
+
+Optional installation of {tekton-triggers} through the Operator::
+* You can choose to not install {tekton-triggers} when deploying {pipelines-shortname} through the Operator. This enhancement provides finer control over component installation, allowing flexibility for environments where triggers are managed separately. The default setting is `false`.
```
When deploying {pipelines-shortname} through the Operator, you can choose not to install {tekton-triggers}. This update provides finer control over component installation and supports environments where triggers are managed independently. The default value is `false`.

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+New flag to disable live collection in tekton-results-watcher::
+* {tekton-results} supports a new flag for the `tekton-results-watcher` controller to disable live collection of runs by setting the following flag in the `TektonConfig` CRD:
++
+[source, yaml]
```
Same as above for a possible addition of a lead-in sentence.

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+Using the request body for sensitive information reduces exposure in logs and improves security. For more details, see link:https://pipelinesascode.com/docs/guide/incoming_webhook/#recommended-post-json-body-method[Recommended (POST JSON body) method].
+
+Detailed logging for GitHub API calls::
+* {pac} supports detailed logging for GitHub API calls, providing insights into API interactions, durations, and rate-limiting. By setting the controller log level to 'debug', you can troubleshoot complex issues more efficiently. This enhancement refactors code to instrument GitHub calls and affects all types of GitHub API calls within the provider. For more details, see link:https://pipelinesasc ...
```
For more details -> For more information.

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+* {pac} adds the `auto-configure-repo-repository-template` setting, a new option in `auto-configure-new-github-repo` feature setting to consolidate repository creation within a single namespace, streamlining management for those with multiple repositories. For more details, see "Additional resources".
+
+Support for relative task references in remote Pipeline definitions::
+* You can use relative paths to reference tasks from within a remote `Pipeline` definition. The {pac} resolver automatically builds the full URL for the task based on the location of the remote `PipelineSpec` definition. This improves pipeline organization and portability by keeping tasks and pipelines co-located, which is especially useful when working with tagged releases or branches. For more  ...
```
You can now use relative paths to reference tasks within a remote `Pipeline` definition. The {pac} resolver automatically constructs/builds the entire/complete URL for the task based on the location of the remote `PipelineSpec` definition. This enhancement improves pipeline organization and portability, particularly when working with tagged releases or branches. For more information, see link:https://pipelinesascode.com/docs/guide/resolver/#overriding-tasks-from-a-remote-pipeline-on-a-pipelinerun[Overriding tasks from a remote pipeline on a PipelineRun].

Dhruv-Soni11 commented on modules/op-release-notes-1-20-0.adoc:
```diff
@@ -0,0 +1,289 @@
+[id="pipelines-breaking-changes-1-20-0_{context}"]
+=== Breaking changes
+
+* The public instance of Tekton Hub (`hub.tekton.dev`) is deprecated and will be removed in a future release. For more details, see "Additional resources".
```
Check earlier feedback for more details, see "Additional Resources".

ochromy committed:
Merge branch 'pipelines-docs-main' into RHDEVDOCS-6657
